src/americana.js

```diff
@@ -255,20 +255,38 @@ function buildLayers() {
     lyrPlace.continent
   );
 
-  // Resolve localized name placeholders.
+  // Updated localizable variables at the top level of each layer's text-field expression based on the current language preference.
+  let locale = Label.getLocales()[0];
   let localizedNameExpression = Label.getLocalizedNameExpression(false);
   let legacyLocalizedNameExpression = Label.getLocalizedNameExpression(true);
   for (let layer of layers) {
-    if (
-      "metadata" in layer &&
-      "layout" in layer &&
-      layer.metadata["americana:text-field-localized"] === true
-    ) {
-      // https://github.com/openmaptiles/openmaptiles/issues/769
-      layer.layout["text-field"] =
-        layer["source-layer"] === "transportation_name"
-          ? legacyLocalizedNameExpression
-          : localizedNameExpression;
+    if ("layout" in layer && "text-field" in layer.layout) {
+      let textField = layer.layout["text-field"];
+      if (textField && textField[0] === "let") {
+        if (textField.indexOf("localizedName") % 2 === 1) {
+          let variableNameIndex = textField.indexOf("localizedName");
+          if (variableNameIndex % 2 === 1) {
+            // https://github.com/openmaptiles/openmaptiles/issues/769
+            let expr =
+              layer["source-layer"] === "transportation_name"
+                ? legacyLocalizedNameExpression
+                : localizedNameExpression;
+            layer.layout["text-field"][variableNameIndex + 1] = expr;
+          }
+        }
+        if (textField.indexOf("localizedCollator") % 2 === 1) {
+          let variableNameIndex = textField.indexOf("localizedCollator");
+          if (variableNameIndex % 2 === 1) {
+            let options = {
+              "case-sensitive": false,
+              "diacritic-sensitive": true,
```

Comment:
The Rand McNally policy seems good.  Showing two names varying only by diacritics looks fairly redundant.  The Quebec City example looks like it maybe should be changed in Wikidata.   Seems like the official name of the city may be Québec in both French and English.  Although the form Quebec City is common, it appears to be informal usage.
https://www.btb.termiumplus.gc.ca/tpv2guides/guides/wrtps/index-eng.html?lang=eng&lettr=indx_catlog_q&page=9_zoRYLgt2AY.html

Reply:
I'm in favor of folding diacritics in English. In such cases, the diacritic-free English name is clearly just derived from the local name, and it feels weird to show both.

I'm not as supportive of removing whole words. I'd rather keep "City" in there.

Reply:
If you look at the the list of all languages on the [Quebec City Wikidata item](https://www.wikidata.org/wiki/Q2145), you'll see the forms "Quebec" and "Quebec City" assigned to different languages.  I doubt this represents what a speaker of those languages would truly expect and more just a reflection that both forms are used interchangeably.  Either would look fine on a map to me, but "Québec (Quebec City)" or "Quebec City (Québec)" seems redundant and not helpful.  [New York City](https://www.wikidata.org/wiki/Q60) has the same issue.  Languages are split between "New York" and "New York City".  To avoid the redundancy it seems reasonable to me that we would consider "Name" and "Name City" equivalent for the purpose of deciding if a gloss is needed or not.

Reply:
> To avoid the redundancy it seems reasonable to me that we would consider "Name" and "Name City" equivalent for the purpose of deciding if a gloss is needed or not.

Should each `text-field` property hard-code the word “City”? What about when the preferred language is something other English – should the property also hard-code “Ciudad de” in Spanish, etc.?

Reply:
Yes I suppose ideally "{{NAME}}", "{{NAME}} City", "Ciudad de {{NAME}}", and similar constructions in other languages would all be considered equivalent.  Could become quite a long list though 😳.  Maybe it's not worth it.  Then of course you have to wonder if Nueva York is different enough from New York that it should get a gloss...

Reply:
I tried a simple approach to substituting the local name when it matches the localized name after diacritic folding. It does reduce redundancy all over the world when the preferred language is English. It looks good:

Before | After
----|----
[<img src="https://user-images.githubusercontent.com/1231218/204693817-beb6a794-1287-40bc-88cd-c1418b20b954.png" width="400" alt="Montreal">](https://zelonewolf.github.io/openstreetmap-americana/#map=5/46.32/-72.2&language=en) | [<img src="https://user-images.githubusercontent.com/1231218/204693834-d5d3d18d-6221-449a-bcdc-62784de98af4.png" width="400" alt="Montréal">](https://zelonewolf.github.io/openstreetmap-americana/#map=5/46.32/-72.2&language=en)
[<img src="https://user-images.githubusercontent.com/1231218/204694683-ec57a21a-96a1-4edb-b68b-e0689c41ce0c.png" width="400" alt="Cancun">](https://zelonewolf.github.io/openstreetmap-americana/#map=11/21.1473/-86.8055&language=en) | [<img src="https://user-images.githubusercontent.com/1231218/204694692-303acfb4-72cf-4220-94c3-dcfa3bbdbc6e.png" width="400" alt="Cancún">](https://zelonewolf.github.io/openstreetmap-americana/#map=11/21.1473/-86.8055&language=en)

However, this approach will actually remove diacritics when the user prefers Spanish and looks at an English-speaking country. Meanwhile, French and Spanish use the acute accent for different purposes, so cities in French-speaking regions just look misspelled:

Before | After
----|----
[<img src="https://user-images.githubusercontent.com/1231218/204695631-ce5ec2c9-d2a8-4b70-84d1-5afb39bc1b9b.png" width="400" alt="Los Ángeles">](https://zelonewolf.github.io/openstreetmap-americana/#map=6/34.019/-118.223&language=es) | [<img src="https://user-images.githubusercontent.com/1231218/204695645-b5296001-9111-4a35-bffe-4460b3709f9b.png" width="400" alt="Los Angeles">](https://zelonewolf.github.io/openstreetmap-americana/#map=6/34.019/-118.223&language=es)
[<img src="https://user-images.githubusercontent.com/1231218/204697434-8f3b9f16-56bf-459c-912c-28bf884633c2.png" width="400" alt="Montreal">](https://zelonewolf.github.io/openstreetmap-americana/#map=5/46.08/-72.06&language=es) | [<img src="https://user-images.githubusercontent.com/1231218/204697439-ba60a495-a5b0-4494-85cd-dd8dfcb1b8bb.png" width="400" alt="Montréal">](https://zelonewolf.github.io/openstreetmap-americana/#map=5/46.08/-72.06&language=es)

A robust solution to this problem would require GL JS to support setting the collator’s [`sensitivity`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator) option, and we’d probably need to know the local language to apply both languages’ collators: https://github.com/ZeLonewolf/openstreetmap-americana/pull/592#discussion_r1035286767. However, I’m inclined to limit the diacritic folding to when the user prefers English, a language that natively uses relatively few diacritics where we’re certain that map publishers place non-English diacritics on foreign place names. I’ve added a special case for English; if we learn of another language that also uses diacritics mainly for foreign place names, we can add it to the special case.

Ironically, if the user prefers English and looks at Vietnam, we’d prefer the spelling used in Vietnam over the spelling used by the overseas community, including in the U.S., but either OSM or Wikidata would need `vi-US` or `vi-VN` tagging in order for us to solve that problem.

Before | After
----|----
[<img src="https://user-images.githubusercontent.com/1231218/204692233-3cafc4a2-a222-4c17-b3ed-7619a30b4628.png" width="400" alt="Da Nang">](https://zelonewolf.github.io/openstreetmap-americana/#map=9/15.921/108.1409&language=en) | [<img src="https://user-images.githubusercontent.com/1231218/204692252-7726b4fd-2ec0-4340-99de-3ea91bbfc819.png" width="400" alt="Đà Nẵng">](https://zelonewolf.github.io/openstreetmap-americana/#map=9/15.921/108.1409&language=en)
[<img src="https://user-images.githubusercontent.com/1231218/204691704-64baf0a1-caa3-4e0b-8939-2e8e28362d98.png" width="400" alt="Thanh Hóa">](https://zelonewolf.github.io/openstreetmap-americana/#map=6.84/19.67/105.835&language=en) | [<img src="https://user-images.githubusercontent.com/1231218/204691488-7d37ab2d-77c6-4333-9cd0-1f6f2d24172b.png" width="400" alt="Thanh Hoá">](https://zelonewolf.github.io/openstreetmap-americana/#map=6.84/19.67/105.835&language=en)

Reply:
Man.  Tough problem.  Would it work better to keep the localized name instead, but just omit the local name gloss when they match after diacritic folding?

Reply:
I think as long as we have a local name gloss on any city, the lack of a gloss implies that the remaining name is also the local name. Alternatively, we could put the local name first, as in National Geographic maps. However, that would exacerbate the messy multilingual names in https://github.com/ZeLonewolf/openstreetmap-americana/pull/592#discussion_r1035284310, and some labels would be primarily labeled in a non-Latin script because we don’t have a solution for transliteration yet: https://github.com/ZeLonewolf/openstreetmap-americana/pull/592#discussion_r1035511841.

Reply:
That makes sense.  I just figured that although Montreal/Montréal, Los Angleles/Los Ángeles are different it may not be a big enough difference to warrant a gloss.

Reply:
I see where you’re coming from, but personally I think this gloss feature would be more predictable and intuitive if it omits information only when it’s literally redundant (as in the base letters being the same), rather than when it’s different but not very much so. When I first saw Monterey, California, labeled as “Monterrey <small>(Monterey)</small>” in Spanish, I instinctively thought it was a typo in either OSM or Wikidata, but actually that’s an interesting bit of trivia related to how the Spanish language evolved after California’s statehood.

Reply:
> as in the base letters being the same

If I told you that Á and A look like the same base letter to me, would you consider me an uncultured 'merican? 🤠

Reply:
I would take that as valuable feedback about the fonts we’re using. 😉

Reply:
Oh no, I can _see_ the diacritics.  My English speaking brain just dismisses them as irrelevant details 😅

Reply:
I’m not here to judge. 🙂 You’re likely going to stick to `language=en` except maybe for debugging purposes, so the extra accent mark will only show up in countries where you’ll be seeing accent marks on lots of non-city features anyways. I think it’ll be less jarring once we have a solution for transliteration and multilingual names so that we can flip this feature around per National Geographic, putting the local language on top and an occasional, optional English name below.